### PR TITLE
add TimeTriggerConfig

### PR DESCRIPTION
**Pull request by nerdyyatrice**
add TimeTriggerConfig for java, need to wait for the region that we test against upgraded first.

### CODE REVIEW

File: src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java [hunk 1]
```diff
@@ -1293,6 +1288,77 @@ private void testLogTrigger() throws ParseException {
...
+        CreateTriggerRequest createTReq = new CreateTriggerRequest(SERVICE_NAME, FUNCTION_NAME);
+        TimeTriggerConfig timeTriggerConfig = new TimeTriggerConfig(cronEvery, payload, true);
+        createTReq.setTriggerName(triggerName);
+        createTReq.setTriggerType(TRIGGER_TYPE_TIMER);
+        createTReq.setInvocationRole(INVOCATION_ROLE);
+        createTReq.setTriggerConfig(timeTriggerConfig);
+        CreateTriggerResponse createTriggerResponse = client.createTrigger(createTReq);
+        assertEquals(triggerName, createTriggerResponse.getTriggerName());
+        assertEquals(TRIGGER_TYPE_TIMER, createTriggerResponse.getTriggerType());
+        assertEquals(INVOCATION_ROLE, createTriggerResponse.getInvocationRole());
```
Comment (wanghq) [inline src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java]: ?

File: src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package com.aliyuncs.fc.model;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Time Trigger config
+ */
+public class TimeTriggerConfig {
```
Comment (wanghq) [inline src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java]: Can you follow log trigger configs' convention? e.g. https://github.com/aliyun/fc-java-sdk/blob/aebb97aa8cbefc28c622b1f8bda876c9cdcad6d7/src/main/java/com/aliyuncs/fc/model/LogTriggerConfig.java

if I just want to disable trigger, I don't need to set `cronExpression` and `payload` as below, I can just do `new TimeTriggerConfig().SetEnable(false)`.

`updateReq.setTriggerConfig(new TimeTriggerConfig(cronExpression, payload,true));`

File: src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    @SerializedName("cronExpression")
+    private String cronExpression;
+
+    @SerializedName("payload")
+    private String payload;
+
+    @SerializedName("enable")
+    private boolean enable;
```
Comment (wanghq) [inline src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java]: This should be Boolean type so the default value is null and won't be included in the payload if it's not set.

File: src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    public void setEnable(boolean enable) {
+        this.enable = enable;
+    }
+
+    public String getPayload() {
+        return payload;
+    }
+
+    public void setPayload(String payload) {
```
Comment (shuaichang) [inline src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java]: Should payload be byte[] payload ?

File: src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java [hunk 5]
```diff
@@ -111,6 +107,7 @@
     private static final String TRIGGER_NAME = "testTrigger";
     private static final String TRIGGER_TYPE_OSS = "oss";
     private static final String TRIGGER_TYPE_LOG = "log";
+    private static final String TRIGGER_TYPE_TIMER = "timer";
```
Comment (shuaichang) [inline src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java]: Seems we call the trigger "time" trigger everywhere, why we make the type "timer" here?

Comment (nerdyyatrice) [reply src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java]: the type of the trigger is called "timer" but the service is called "time trigger"

File: src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java (hunk 4 above)
Comment (nerdyyatrice) [reply src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java]: it's defined as a string in our protocol
	Payload        *string `json:"payload"`

File: src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public String getCronExpression() {
+        return cronExpression;
+    }
+
+    public TimeTriggerConfig setCronExpression(String cronExpression) {
+        this.cronExpression = cronExpression;
+        return this;
+    }
+
+    public boolean isEnable() {
```
Comment (wanghq) [inline src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java]: boolean -> Boolean

File: src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        return enable;
+    }
+
+    public TimeTriggerConfig setEnable(boolean enable) {
```
Comment (wanghq) [inline src/main/java/com/aliyuncs/fc/model/TimeTriggerConfig.java]: boolean -> Boolean

File: src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java [hunk 8]
```diff
@@ -1293,6 +1288,75 @@ private void testLogTrigger() throws ParseException {
...
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+        }
+
+        // Update Triggers
+        UpdateTriggerRequest req = new UpdateTriggerRequest(SERVICE_NAME, FUNCTION_NAME, triggerName);
+        req.setTriggerConfig(new TimeTriggerConfig().setCronExpression(cronExpression).setPayload(payload).setEnable(true));
+        UpdateTriggerResponse updateTResp = client.updateTrigger(req);
+        assertEquals(triggerOld.getTriggerName(), updateTResp.getTriggerName());
+        assertEquals(triggerOld.getInvocationRole(), updateTResp.getInvocationRole());
```
Comment (wanghq) [inline src/test/java/com/aliyuncs/fc/FunctionComputeClientTest.java]: Why checking the role?

**Pull request merged**